webpack.config.js

```diff
@@ -1,8 +1,13 @@
 const webpack = require( 'webpack' );
 const defaultConfig = require( '@wordpress/scripts/config/webpack.config' );
+const { hasArgInCLI } = require( '@wordpress/scripts/utils' );
```

Comment:
❓ Maybe dumb question. Not a super expert in Webpack. But why we are loading all this logic in `webpack.config.js` if we are using `webpack-development.config.js`?

Reply:
Sorry, I do not quite understand this question. Could you provide more context?

Reply:
Yes, so, we have `webpack.config.js` in which we have all our project config. There was some code added there in this PR. 

Then I see in package.json we use `"start:hot": "npm run dev -- --config ./webpack-development.config.js && npm start -- --hot --allowed-hosts all",`

We call `./webpack-development.config.js` however we still loaded code in `webpack.config.js`. 

My doubt is why we are loading new code regarding fast refresh in `webpack.config.js` if we are actually loading `./webpack-development.config.js` for that.

Reply:
The React Fast Refresh is supposed to be bundled with vendor libraries together and initialized/loaded **before** React. But upon the WP architecture, vendor libraries are bundled separately and maybe loaded via DEWP, so the React Fast Refresh needs to be enabled in an [externalizing way](https://github.com/pmmmwh/react-refresh-webpack-plugin/blob/main/docs/TROUBLESHOOTING.md#externalising-react).

It's the same In GLA, so the `webpack-development.config.js` is for building the React Fast Refresh code externally and also mixing the building flow of DEWP, and then using the js dependency and loading mechanism of WP to ensure they will be loaded before React.

While it is possible to create separate js files by adding those `entry` to `webpack.config.js`, sharing the `plugins` of webpack config will make the React Fast Refresh files unable to use different DEWP initial settings.

Exporting two webpack configs as an array from `webpack.config.js` will cause the built React Fast Refresh js files to be deleted by [another webpack plugin](https://github.com/WordPress/gutenberg/blob/%40wordpress/scripts%4022.1.0/packages/scripts/config/webpack.config.js#L226-L231), `CleanWebpackPlugin`, after saving GLA source code changes because the plugins are not shared between the two webpack configs. The root cause of the unexpected deletion with `CleanWebpackPlugin` can be found in https://github.com/woocommerce/pinterest-for-woocommerce/issues/330.

About the `ReactRefreshWebpackPlugin`, it needs to work with the app codes, so it's placed in the `webpack.config.js`.